src/main/java/com/foundrysoftware/minecraft/plugins/PluginTemplate.java

```diff
@@ -34,122 +34,135 @@
-                                }
-                                if(isArrow(item)==true){
-                                    String potion = args[0];
-                                    item.setType(Material.TIPPED_ARROW);
-                                    item.setAmount(item.getAmount());
-                                    PotionMeta potionMeta = (PotionMeta)item.getItemMeta();
-                                    potionMeta.setBasePotionData(new PotionData(PotionType.getByEffect(PotionEffectType.getByName(potion))));
-                                    int duration =10;
-                                    int amplifier =2;
-                                    boolean visible=true;
-                                    potionMeta.addCustomEffect(new PotionEffect(PotionEffectType.getByName(potion),duration,amplifier),visible);
-                                    item.setItemMeta(potionMeta);
                     }
-
                 }
             }
-        }}}
+        }
         return false;
     }
+
     //@Override
     @EventHandler
-    public boolean onShoot(ProjectileHitEvent e){
-       if(e.getEntity() instanceof Arrow) {
-           Arrow arrow = (Arrow) e.getEntity();
-           Entity shooter = (Entity) arrow.getShooter();
-           Location loc;
-           if (e.getHitBlock() != null) {
-               loc = e.getHitBlock().getLocation();
-           } else {
-               loc = e.getHitEntity().getLocation();
-           }
-           Bukkit.getServer().broadcastMessage(loc.toString());
-           if (arrow.hasCustomEffect(PotionEffectType.CONDUIT_POWER)) {
-               loc.getWorld().createExplosion(loc,10,false,false,shooter);//location of explosion,size,setFire,DoBlockDamage,ExplosionOwner
-           }
-           //if(arrow.hasCustomEffect(PotionEffectType.DOLPHINS_GRACE)){
-              // for(Entity ent: loc.getWorld().getNearbyEntities(loc,5,5,5)){
-                   //if(ent instanceof LivingEntity){
-                      // (Damageable)((LivingEntity) ent).damage(50, shooter);
-                   //}
-              // }
-           return true;
-           }
-           return false;
-       }
-       public boolean isArrow(ItemStack item){
-        if(item.getType()== Material.TIPPED_ARROW|| item.getType() == Material.ARROW || item.getType()==Material.SPECTRAL_ARROW){
+    public boolean onShoot(ProjectileHitEvent e) {
+        if (e.getEntity() instanceof Arrow) {
+            Arrow arrow = (Arrow) e.getEntity();
+            Entity shooter = (Entity) arrow.getShooter();
+            Location loc=getLocation(e);
+            Bukkit.getServer().broadcastMessage(loc.toString());
+            if (arrow.hasCustomEffect(PotionEffectType.CONDUIT_POWER)) {
+                loc.getWorld().createExplosion(loc, 10, false, false, shooter);//location of explosion,size,setFire,DoBlockDamage,ExplosionOwner
+            }
+            //if(arrow.hasCustomEffect(PotionEffectType.DOLPHINS_GRACE)){
```

Comment:
Issue 3#) Remove ALL comments whether code or otherwise (aka 123,125-130. Please check all the code and remove all occurrences of commetns